### DISCUSSION TITLE
Get the type of an object using Python API

### DISCUSSION THREAD

**Discussion by gmermoud** [geldata/gel#5341] [Q&A] [answered]
I am struggling a bit with the Python API. Say I have an abstract type X, and two derived types A and B. Then, the following query:

```
select X {
 field1,
 field2
}
```
returns a list of [`edgedb.Object`](https://www.edgedb.com/docs/clients/python/api/types#edgedb.Object), but it gives no clue as to which is the type of the object. Is there a way around this limitation?

**Comment by raddevon** [comment, marked as answer]
You can add the `__type__` link to your shape to get type information:

```
select X {
  field1,
  field2,
  __type__ {
    name
  }
}
```

**Reply by gmermoud** [reply]
Wow, fantastic. Thanks a lot for this, it feels like EdgeDB can do everything 😄
